dness: init at 0.5.7

LoganBarnett opened the pull request:
This adds the `dness` program.  `dness` is a dynamic DNS tool capable of updating multiple DNS providers.

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [Nixpkgs 25.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/doc/release-notes/rl-2511.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/doc/manual/release-notes/rl-2505.section.md) Nixpkgs Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
- [NixOS 25.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2511.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) NixOS Release notes)
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

Please note I am a new contributor - to the best of my knowledge, this fits both the guidelines and the patterns I've observed in other packages.  Any deviation is just something I missed, and I am happy to correct :)

I have tested what I believe is the main flow of the program - querying the WAN IP and then updating a DNS record with the provider (Porkbun in my case).  You can see my configuration used here: https://github.com/loganbarnett/dotfiles/blob/master/nix/nixos-modules/dns-dynamic-ip-home.nix

If we like this and merge it, I can see about polishing up the above configuration into a proper NixOS module.

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

LoganBarnett committed:
maintainers: add logan-barnett

LoganBarnett committed:
dness: init at 0.5.7

LoganBarnett commented:
Took me a hot minute to figure out what was going on with these `Eval / Outpaths` failing.  I have a bead on it I think.  Apologies for the noise.

LoganBarnett commented:
Alright, all the checks pass.  Ready for review!

yzhou216 commented on pkgs/by-name/dn/dness/package.nix:
```diff
@@ -0,0 +1,46 @@
+      "--skip=namecheap::tests::test_namecheap_update"
+      "--skip=noip::tests::test_noip_update"
+    ];
+    meta = {
+      description = "A dynamic DNS updating tool supporting a variety of providers.";
+      homepage = "https://github.com/nickbabcock/dness";
+      maintainers = with maintainers; [ logan-barnett ];
+      platforms = platforms.unix;
+      license = licenses.mit;
+    };
```
```suggestion

  meta = {
    description = "Dynamic DNS updating tool supporting a variety of providers";
    homepage = "https://github.com/nickbabcock/dness";
    maintainers = with lib.maintainers; [ logan-barnett ];
    license = lib.licenses.mit;
  };
```

yzhou216 commented on pkgs/by-name/dn/dness/package.nix:
```diff
@@ -0,0 +1,46 @@
+  openssl,
+  pkg-config,
+  rustPlatform,
+  ...
+}:
+rustPlatform.buildRustPackage (
+  let
+    inherit (lib) licenses maintainers platforms;
+    name = "dness";
+    version = "v0.5.7";
+    src = fetchFromGitHub {
+      owner = "nickbabcock";
+      repo = "dness";
+      rev = version;
+      hash = "sha256-Vty4ec6aoUh3p2b9vLkNeS5R4pJWzjwYrC5DtVVyhT8=";
+    };
+  in
+  {
+    pname = name;
+    inherit src version;
```
```suggestion
}:

rustPlatform.buildRustPackage (finalAttrs: {
  name = "dness";
  version = "0.5.7";
  src = fetchFromGitHub {
    owner = "nickbabcock";
    repo = "dness";
    tag = "v${finalAttrs.version}";
    hash = "sha256-Vty4ec6aoUh3p2b9vLkNeS5R4pJWzjwYrC5DtVVyhT8=";
  };

```

yzhou216 commented on pkgs/by-name/dn/dness/package.nix:
```diff
@@ -0,0 +1,46 @@
+      pkg-config
+    ];
+    PKG_CONFIG_PATH = "${openssl.dev}/lib/pkgconfig";
+    cargoHash = "sha256-VyjntXsb1FUNguJFEmaZmjeCUHPMHMDICTTMDwExNBI=";
```
```suggestion

  cargoHash = "sha256-WhSeNukPjgM7Cy8LWi/s1YGa5/UxsFU1NGL7vIUlU58=";

```

Cargo hash changed for some reason.

yzhou216 reviewed (changes requested):
Cleanup

LoganBarnett commented:
@yzhou216 Thanks for the quick review!  I pushed up more changes with your corrections.  I ran into a problem relating to dropping `pname` in favor of `name` when it came to `versionCheckHook`.  This is the error I got:

```
       > Executing versionCheckPhase
       > both $pname and $versionCheckProgram are empty, so we don't know which program to run the versionCheckPhase upon
```

If I simply use `pname`, over `name` the problem goes away.

I don't have a preference on which direction to go with the fix, but is it critical to favor `name`?  I thought generally `name` was a computed composition of `pname` and `version`, with the possibility of being overwritten by the derivation author.

yzhou216 commented:
> @yzhou216 Thanks for the quick review! I pushed up more changes with your corrections. I ran into a problem relating to dropping `pname` in favor of `name` when it came to `versionCheckHook`. This is the error I got:
> 
> ```
>        > Executing versionCheckPhase
>        > both $pname and $versionCheckProgram are empty, so we don't know which program to run the versionCheckPhase upon
> ```
> 
> If I simply use `pname`, over `name` the problem goes away.
> 
> I don't have a preference on which direction to go with the fix, but is it critical to favor `name`? I thought generally `name` was a computed composition of `pname` and `version`, with the possibility of being overwritten by the derivation author.

Yes, `pname` should be used here.  Sorry, I didn't catch that when I was reviewing.  I think `versionCheckHook` looks for `pname` and `meta.mainProgram` for the executable name by default.

LoganBarnett commented on pkgs/by-name/dn/dness/package.nix:
```diff
@@ -0,0 +1,32 @@
+  meta = {
+    description = "Dynamic DNS updating tool supporting a variety of providers";
+    homepage = "https://github.com/nickbabcock/dness";
+    maintainers = with lib.maintainers; [ logan-barnett ];
```
This is done, but GitHub isn't marking the section as outdated.

LoganBarnett commented on pkgs/by-name/dn/dness/package.nix:
```diff
@@ -0,0 +1,32 @@
+}:
+rustPlatform.buildRustPackage (finalAttrs: {
+  name = "dness";
+  version = "0.5.7";
```
This is done, but GitHub isn't marking the section as outdated.

LoganBarnett commented:
@yzhou216 all of the items you have should be resolved now.  Anything else you want to see done? :)

yzhou216 commented:
> @yzhou216 all of the items you have should be resolved now. Anything else you want to see done? :)

Please update the commit message and PR title to be `dness: init at 0.5.7`.

yzhou216 reviewed (approved):
> @yzhou216 finally got around to the rename. Apologies for the delay.

No need to apologize.  Thanks for the contribution!

yzhou216 commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 411007`
Commit: [`c156dd0dab641c6903bed9afa39d0f86cc653236`](https://github.com/NixOS/nixpkgs/commit/c156dd0dab641c6903bed9afa39d0f86cc653236) ([subsequent changes](https://github.com/NixOS/nixpkgs/compare/c156dd0dab641c6903bed9afa39d0f86cc653236..pull/411007/head))
Merge: [`d2cc35dba64617cab1203b6f700039b00badea1a`](https://github.com/NixOS/nixpkgs/commit/d2cc35dba64617cab1203b6f700039b00badea1a)

Logs: https://github.com/yzhou216/nixpkgs-review-gha/actions/runs/17997586876

<details><summary>Download packages from cache:</summary><ul>
<li><details><summary><code>x86_64-linux</code></summary>

```shell
nix-store -r --add-root nixpkgs-pr-411007-x86_64-linux \
  --option binary-caches 'https://cache.nixos.org/ https://yzhou216.cachix.org' \
  --option trusted-public-keys '
  cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDShjY=
  yzhou216.cachix.org-1:O7nGwQnLGGBGE8zYFMipVzPVN956FcV57y9SqwEP+O8=
  ' \
  /nix/store/w4gj3304bly5c27cb51cr8ay6nljgmyq-dness-0.5.7
```
</details></li>
<li><details><summary><code>aarch64-linux</code></summary>

```shell
nix-store -r --add-root nixpkgs-pr-411007-aarch64-linux \
  --option binary-caches 'https://cache.nixos.org/ https://yzhou216.cachix.org' \
  --option trusted-public-keys '
  cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDShjY=
  yzhou216.cachix.org-1:O7nGwQnLGGBGE8zYFMipVzPVN956FcV57y9SqwEP+O8=
  ' \
  /nix/store/3ac06bc0i5y6ixj3k6pbxdxp5609h3g8-dness-0.5.7
```
</details></li>
<li><details><summary><code>x86_64-darwin</code></summary>

```shell
nix-store -r --add-root nixpkgs-pr-411007-x86_64-darwin \
  --option binary-caches 'https://cache.nixos.org/ https://yzhou216.cachix.org' \
  --option trusted-public-keys '
  cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDShjY=
  yzhou216.cachix.org-1:O7nGwQnLGGBGE8zYFMipVzPVN956FcV57y9SqwEP+O8=
  ' \
  /nix/store/kd74jcr6nm6zp29h6zjbwchiqb0mam01-dness-0.5.7
```
</details></li>
<li><details><summary><code>aarch64-darwin</code></summary>

```shell
nix-store -r --add-root nixpkgs-pr-411007-aarch64-darwin \
  --option binary-caches 'https://cache.nixos.org/ https://yzhou216.cachix.org' \
  --option trusted-public-keys '
  cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDShjY=
  yzhou216.cachix.org-1:O7nGwQnLGGBGE8zYFMipVzPVN956FcV57y9SqwEP+O8=
  ' \
  /nix/store/29kdcvsvhg3s7kg03hvlkmhd8d7f6q3d-dness-0.5.7
```
</details></li>
</ul></details>


---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>dness</li>
  </ul>
</details>


---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>dness</li>
  </ul>
</details>


---
### `x86_64-darwin` (sandbox = true)
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>dness</li>
  </ul>
</details>


---
### `aarch64-darwin` (sandbox = true)
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>dness</li>
  </ul>
</details>

LoganBarnett commented:
I rebased atop master and corrected the conflict with the `maintainers-list.nix` file.  What are the chances I'd have to do that twice in the same PR?!  (across many months... 😅 )

SuperSandro2000 commented:
It does not matter if GitHub marks the code point as outdated, please just mark the conversation as resolved 😅 Thanks
